packages/editor/src/components/PreviewDataTable.tsx

```diff
@@ -71,37 +71,36 @@ const PreviewDataTable = () => {
 
   const dispatch = useContext(EditorDispatchContext)
 
-  const fetchAsyncData = async () => {
-    if (config.type === 'dashboard') {
-      Object.keys(config.datasets).forEach(async datasetKey => {
-        if (config.datasets[datasetKey].preview) {
-          if (config.datasets[datasetKey].dataUrl) {
-            const remoteData = await fetchRemoteData(config.datasets[datasetKey].dataUrl)
-            if (Array.isArray(remoteData)) {
-              setTableData(remoteData)
+  useEffect(() => {
+    const fetchAsyncData = async () => {
```

Comment:
why is it necessary to move this function into the useEffect?

Reply:
The function gets re created each time dependancy changes. if it is outside it will not. we have issue on data preview, it does not gets shown and if we trigger other parts of page it comes.Just thought it will be helpfull to fix a bug.Any other ideas ?

Reply:
I think you could maybe pass in `datasets` into the fetchAsyncData function